### ISSUE TITLE
package not installing for ubunut 16.04

### ISSUE THREAD

**Issue by hrani** [BhallaLab/moose-package#1]
I tried to install moose package as per instruction from here for 16.04
 https://software.opensuse.org/download.html?project=home:moose&package=moose
 sudo apt-get install moose
This is fine 
when I tried typing moose in the terminal it say
jalaja@jalaja-Inspiron-3442:~$ Traceback (most recent call last):
  File "mgui.py", line 59, in <module>
    import mplugin
  File "/usr/lib/moose/gui/mplugin.py", line 49, in <module>
    import moose
  File "/usr/lib/python2.7/dist-packages/moose/__init__.py", line 25, in <module>
    from .moose import *
  File "/usr/lib/python2.7/dist-packages/moose/moose.py", line 20, in <module>
    import genesis.writeKkit
  File "/usr/lib/python2.7/dist-packages/moose/genesis/__init__.py", line 1, in
<module>
    from  .writeKkit import mooseWriteKkit
  File "/usr/lib/python2.7/dist-packages/moose/genesis/writeKkit.py", line 16, in
<module>
    from moose.chemUtil.chemConnectUtil import *
ImportError: No module named chemUtil.chemConnectUtil

check this.

**Comment by dilawar** [comment]
Can you provide the python version? Is it 2.7.12 or 2.7.9?

**Comment by dilawar** [comment]
Confirmed. Failing on debian-8 (ubuntu-16.04) due to missing entry in `setup.py` file. Added `moose.chemUtil` folder in setup.py file on chamcham branch.

**Comment by dilawar** [comment]
Revision 88 on https://build.opensuse.org/package/show/home:moose/moose should fix it.

**Comment by hrani** [comment]
Nope still the same
got sudo dpkg -i moose_3.1.2_amd64.deb
and import moose gives
Python 2.7.12 (default, Nov 19 2016, 06:48:10) 
[GCC 5.4.0 20160609] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import numpy
>>> import moose
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python2.7/dist-packages/moose/__init__.py", line 25, in <module>
    from .moose import *
  File "/usr/lib/python2.7/dist-packages/moose/moose.py", line 20, in <module>
    import genesis.writeKkit
  File "/usr/lib/python2.7/dist-packages/moose/genesis/__init__.py", line 1, in <module>
    from  .writeKkit import mooseWriteKkit
  File "/usr/lib/python2.7/dist-packages/moose/genesis/writeKkit.py", line 16, in <module>
    from moose.chemUtil.chemConnectUtil import *
ImportError: No module named chemUtil.chemConnectUtil

**Comment by dilawar** [comment]
The new build log has the relevant file 

```
[  950s] -- Installing: /usr/src/packages/BUILD/debian/tmp/usr/lib//python2.7/site-packages/moose/moose_constants.py
[  950s] -- Installing: /usr/src/packages/BUILD/debian/tmp/usr/lib//python2.7/site-packages/moose/chemUtil
[  950s] -- Installing: /usr/src/packages/BUILD/debian/tmp/usr/lib//python2.7/site-packages/moose/chemUtil/chemConnectUtil.py
[  950s] -- Installing: /usr/src/packages/BUILD/debian/tmp/usr/lib//python2.7/site-packages/moose/chemUtil/add_Delete_ChemicalSolver.py
[  950s] -- Installing: /usr/src/packages/BUILD/debian/tmp/usr/lib//python2.7/site-packages/moose/chemUtil/__init__.py
[  950s] -- Installing: /usr/src/packages/BUILD/debian/tmp/usr/lib//python2.7/site-packages/moose/chemUtil/graphUtils.py
````
However the available package in repo still does not have the these files.

```
$ sudo dpkg --contents moose*.deb | grep chemUtils
````

Probably it takes some time for new package to be mirrored into repository. I'll update you tomorrow on this thread.

**Comment by dilawar** [comment]
The latest package is in repo. The version has been bumped up but build number. It reads 3.1.2-1 for debain/ubuntu instead of 3.1.2 .

It tested with Debian-8.0. Seems to import moose and moogli just fine. Reopen the issue if it doesn't work.

**Comment by hrani** [comment]
Still Problem with installation.

1.Guess the moosegui doesn't have symbolic link, (one need to locate where
moose-gui is and then run )

2. Does python-libsbml lib needs to be installed separately, it doesn't
come up with the package?
one need to install pip and then python-libsbml (This is not mentioned
anywhere!)

3. Merge commit has wrong update

https://github.com/BhallaLab/moose-gui/blob/1ff6dbfe29b68bb4faa517f68340c7f60de2ce24/plugins/kkitOrdinateUtil.py
(My last commit)

 but your merge commit
https://github.com/BhallaLab/moose-gui/blob/1849c1f86ba75408244ec47a2699987c246a03b1/plugins/kkitOrdinateUtil.py

has these lines, line number 238-241 which is wrong (Should not exists
first of all, not sure how you merged the files)
if int( nx.__version__.split('.') ) >= 11:
        position = nx.graphviz_layout( G )
else:
        position = nx.graphviz_layout(G, prog = 'dot')


which is creates problems when loading specific file type, those line
should not exist.

You need to re-package this

Harsha

**Comment by hrani** [comment]
Re-opening as there is issue with the files

**Comment by dilawar** [comment]
`moosegui` is renamed to just `moose`. There should be at least one binary same as package name. I can add `moosegui` as proxy to `moose` as well.

The branch `chamcham` is used for packaging. Whatever is available on `chamcham` branch of `moose` repo (https://github.com/Bhallalab/moose) is packaged.

**Comment by dilawar** [comment]
```
if int( nx.__version__.split('.') ) >= 11:
    position = nx.graphviz_layout( G ) 
else: 
   position = nx.graphviz_layout(G, prog = 'dot')
```
exists because of program needs to be specified for networkx version less that 1.11. I could not find any issue liked to commit. This is what I could recall from memory. 

Reverted back to your commit.

**Comment by dilawar** [comment]
`python-libsbml` needs to installed manually. Its not in any package repository. Only pip can be used to install it. Not sure if `python-libsml` package is available for Ubuntu-16.04. I guess `import moose ` statement prints that `libsbml` is not found and spit out instructions to install it. 

`python-libsbml` can be in package dependencies.

**Comment by dilawar** [comment]
https://build.opensuse.org/package/show/home:moose/moose?rev=8 fixes the following

- the kkitOrdinateUtil.py file is reverted to commit https://github.com/BhallaLab/moose-gui/blob/1ff6dbfe29b68bb4faa517f68340c7f60de2ce24/plugins/kkitOrdinateUtil.py 
- No change in `python-libsml` dependencies. When not found, a statement is printed onto console when `import moose; moose.mooseReadSBML( 'a.xml', '/model')` is executed

```In [4]: moose.mooseReadSBML( 'a.xml', '/a' )
No python-libsbml found.
This module can be installed by following command in terminal:
	 easy_install python-libsbml
	 apt-get install python-libsbml
Out[4]: <moose.Shell: id=0, dataIndex=0, path=/>
```
- No change in `moosegui` status. It has been renamed to `moose`.

**Comment by hrani** [comment]
when I fix some thing plz leave it as its. This was not required as I had
found a fix by correctly importing the right library.

**Comment by hrani** [comment]
On Mon, May 22, 2017 at 3:12 PM, Dilawar Singh <notifications@github.com>
wrote:

> python-libsbml needs to installed manually. Its not in any package
> repository. Only pip can be used to install it. Not sure if python-libsml
> package is available for Ubuntu-16.04.
>
On jalaja system when I install python-libsbml it installed without any
issue and also I could read SBML files, so guess its available for Ubuntu
16.04 and the same is working on my system which is again 16.04 with 64 bit
system.


> I guess import moose statement prints that libsbml is not found and spit
> out instructions to install it.
>
> python-libsbml can be in package dependencies.
>
No at-least on new system (jalaja system)  with in python terminal
when I
>import moose,
it did not warn or spit out any instruction to install python-libsbml!
I am not sure if this is included with your recent fix but today morning
when I installed through dpkg it didn't say so.
Anyway once your package is complete let me know I will test on any virtual
box the GUI part by myself.

And already mentioned by Upi, I think moose and moosegui should be separate
otherwise it would be confusing
Harsha

**Comment by dilawar** [comment]
My pleasure!

--

**Comment by dilawar** [comment]
`moose -> moosegui` in new update.

`import moose; moose.readSbml( 'a.xml', '/model' )` should spit out instruction to install libsbml if  not found.

**Comment by dilawar** [comment]
Reopen it or create a new ticket if there is any other issue while testing package.
